pkg/agent/openflow/pipeline.go

```diff
@@ -1723,16 +1811,26 @@ func (f *featureNetworkPolicy) conjunctionActionFlow(conjunctionID uint32, table
 
 // conjunctionActionDenyFlow generates the flow to mark the packet to be denied (dropped or rejected) if policyRuleConjunction
 // ID is matched. Any matched flow will be dropped in corresponding metric tables.
-func (f *featureNetworkPolicy) conjunctionActionDenyFlow(conjunctionID uint32, table binding.Table, priority *uint16, disposition uint32, enableLogging bool) binding.Flow {
+func (f *featureNetworkPolicy) conjunctionActionDenyFlow(conjunctionID uint32, table binding.Table, priority *uint16,
+	disposition uint32, enableLogging, isIngress, isIGMP bool) binding.Flow {
 	ofPriority := *priority
 	metricTable := IngressMetricTable
 	tableID := table.GetID()
+	// Ingress rule is not enforced for multicast traffic, and egress rule enforcement for IGMP packets is handled by packetIn.
+	// So, no flows will be generated for the two cases.
+	if f.isMulticastIngressRule(tableID, isIngress, isIGMP) || f.isIGMPEgressRule(tableID, isIGMP, isIGMP) {
+		return nil
+	}
 	if _, ok := f.egressTables[tableID]; ok {
 		metricTable = EgressMetricTable
 	}
-
-	flowBuilder := table.BuildFlow(ofPriority).
-		MatchConjID(conjunctionID).
+	flowBuilder := table.BuildFlow(ofPriority)
+	if f.isMulticastEgressRule(tableID, isIngress, isIGMP) {
+		metricTable = MulticastEgressMetricTable
+	} else if f.isIGMPIngressRule(tableID, isIngress, isIGMP) {
+		metricTable = MulticastIngressMetricTable
+	}
+	flowBuilder = flowBuilder.MatchConjID(conjunctionID).
```

Comment:
no point to break flowBuilder by metricTable assignment.

Reply:
done

Reply:
done